src/Donations/Micros/PlanetCashSelector.tsx

```diff
@@ -0,0 +1,155 @@
+
+    if (
+      country === profile!.planetCash.country &&
+      paymentSetup.unitCost * quantity >
+        profile!.planetCash.balance / 100 +
+          profile!.planetCash.creditLimit / 100
+    ) {
+      setIsPlanetCashActive(false);
+    }
+  }, [paymentSetup.unitCost, quantity, setIsPlanetCashActive]);
+
+  useEffect(() => {
+    // This is done to lock the transaction with PlanetCash in a single currency.
+    // Here setting country is important as many countries could have same currency.
+
+    // Setting country in global context changes currency of the selected country.
+    if (isPlanetCashActive) {
+      setcountry(profile!.planetCash.country);
+    }
+  }, [isPlanetCashActive, setcountry]);
+
+  useEffect(() => {
+    // Donation with PlanetCash is a one time process.
+    if (frequency !== "once") {
+      setIsPlanetCashActive(false);
+    }
+  }, [frequency]);
+
+  const shouldPlanetCashDisable = () => {
+    let lowBalance = false;
+    let isOnce = false;
+
+    if (
+      country === profile!.planetCash.country &&
+      paymentSetup.unitCost * quantity >
+        profile!.planetCash.balance / 100 +
+          profile!.planetCash.creditLimit / 100
+    ) {
+      lowBalance = true;
+    }
+
+    // Donation with PlanetCash is a one time payment
+    if (frequency === "once") {
+      isOnce = true;
+    }
+
+    return lowBalance || !isOnce;
+  };
+
+  const disabledReason = () => {
+    let lowBalance = false;
+    let isOnce = false;
+
+    if (
+      country === profile!.planetCash.country &&
+      paymentSetup.unitCost * quantity >
+        profile!.planetCash.balance / 100 +
+          profile!.planetCash.creditLimit / 100
+    ) {
+      lowBalance = true;
+    }
+
+    if (frequency === "once") {
+      isOnce = true;
+    }
+
+    if (lowBalance && !isOnce) {
+      return "Balance is low | Donation with PlanetCash is a one time payment";
+    } else if (lowBalance) {
+      return "Balance is low";
+    } else {
+      return "Donation with PlanetCash is a one time payment";
+    }
+  };
+
+  const handleAddBalance = () => {
+    router.replace({
+      query: { ...router.query, to: profile!.planetCash.account },
+    });
+  };
+
+  return (
+    <div className="planet-cash-selector">
+      <div className="planet-cash-selector-toggle">
+        <div>
+          <p>{t("usePlanetCash")}</p>
+          <p>
+            {t("balance")}&nbsp;
+            <span
+              className={
+                "planet-cash-balance" +
+                (Math.sign(
+                  profile!.planetCash.balance / 100 +
+                    profile!.planetCash.creditLimit / 100
+                ) !== -1
+                  ? "-positive"
+                  : "-negative")
+              }
+            >
+              {(
+                profile!.planetCash.balance / 100 +
+                profile!.planetCash.creditLimit / 100
+              ).toFixed(2)}{" "}
```

Comment:
- [ ] The balance amount is not formatted correctly for German language: 

<img width="431" alt="Bildschirmfoto 2022-03-13 um 10 45 10" src="https://user-images.githubusercontent.com/1532418/158054052-436cf68e-67ea-4940-a6a1-18569d4dc860.png">

Reply:
There is a helper function named `getFormatedCurrency` for format amounts of money.

Reply:
Done